### DISCUSSION TITLE
feature request: retype watch should maintain scroll position

### DISCUSSION THREAD

**Discussion by benkoshy** [retypeapp/retype#405] [Ideas]
When writing the documentation, I want to see immediate feedback on how it looks. I continually save my document. this seems to refresh the browser. But the scroll position on the existing document is lost. more details:

Supposing i'm editing something which requires scrolling, everytime i hit refresh, I have to scroll allllll the way back down to the bottom of the page. that's a pain for me. right now, i'm just using CTRL + F to navigate to the relevant portion. 

it would be nice pleasant if this could be done out of the box. not a big deal if it can't be done etc.

i thought the feedback might be useful.

**Comment by geoffreymcgill** [comment]
When I test a long page sample, the scroll position is maintained during a `retype watch` update.

If the page is refreshing and not maintaining the scroll position, then something else might be going wrong.

Are you saving the `.md` file in an editor, such as VS Code? or using the built-in `Edit` button on each page?

Can you provide a sample `.md` file or better yet, a link to a file in a GitHub repo that reproduces this issue?

**Reply by benkoshy** [reply]
> Are you saving the .md file in an editor, such as VS Code? or using the built-in Edit button on each page?

Yes, I am using sublime text, not VS code. I am not editing using the edit button on each page. I did not know that was even possible.

### Steps to reproduce

Link to repo: https://github.com/benkoshy/pagy/tree/mechanical-changes

1. `retype watch`
2. Navigate your browser to a particularly long page. Scroll down to the bottom.
3. Using your text editor (in my case sublime text), make a change towards the bottom of the `.md` file.
4. The browser refreshes, and the scroll position is lost.
5. This means you have to scroll all the way down to the bottom in order to view the changes.

It's not a pleasant experience for the developer - and can get kinda annoying. I save my file dozens of times every few minutes. and I commit to git when it's ready.

I would classify this as a 'nice to have' -- I can certainly make great retype websites without this 'pain point'. It's a minor one in the scheme of things.

**Reply by geoffreymcgill** [reply]
Which version of Retype are you current using on your local machine? Please run the command `retype --version` to get the version number.

**Reply by benkoshy** [reply]
You are absolutely correct: I had been using an older version of `retype`.  

This is the second time I've wasted your time on this point. 

~~My excuse:~~ I thought had updated to the latest version but it wasn't actually applied ~~because I was careless / sloppy and did not actually check that it was successfully updated~~.

My apologies.

**Reply by geoffreymcgill** [reply]
No worries. Once you upgrade to the latest release, can you confirm `retype watch` is maintaining scroll position on a file save.

**Reply by benkoshy** [reply]
Yes I can confirm that it is working - the scroll position is maintained.
